src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,67 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public abstract class AddCommand extends Command {
+    public static String COMMAND_WORD;
+    protected String taskName;
+
+    public AddCommand() {
+        super();
+    }
+
+    public abstract void checkAndSetParameters(String parameterInput) throws DukeException;
+
+    protected abstract void checkParameters(String parameterInput) throws DukeException;
+
+    protected abstract void setParameters(String parameterInput);
+
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException;
+
+    protected boolean containsTaskDescription(String parameterInput, String separator) {
+        // Trim description from parameterInput
+        String[] splits = parameterInput.split(separator, 2);
+        String description = splits[0];
+
+        // Check if description is empty or only contain whitespaces
+        description = description.replaceAll("\\s+", "");
+        if (description.equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected boolean containsTaskTime(String parameterInput, String separator) {
+        // Trim description from parameterInput
+        String[] splits = parameterInput.split(separator, 2);
+        String time = splits[1];
+
+        // Check if description is empty or only contain whitespaces
+        time = time.replaceAll("\\s+", "");
+        if (time.equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    protected boolean containsTimeSeparator(String parameterInput, String separator) {
+        if (!parameterInput.contains(separator)) {
+            return false;
+        } else {
+            return true;
+        }
+    }
```

Comment:
Can be simplified to `parameterInput.contains(separator)`

Reply:
A new check is conducted to mitigate a corner case. 
The suggestion was adopted and combined with the new changes to the code:

```
startsFromTimeSeparator = isStartedFromTimeSeparator(parameterInput, separator);
hasTimeSeparator = parameterInput.contains(separator);

return startsFromTimeSeparator || hasTimeSeparator;
```